gammapy/utils/tests/test_units.py

```diff
@@ -1,8 +1,30 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
-from gammapy.utils.units import standardise_unit
-
+from gammapy.utils.units import standardise_unit, energy_unit_format, energy_str_formatting
+import astropy.units as u
+import numpy as np
+from gammapy.maps import MapAxis
 
 def test_standardise_unit():
     assert standardise_unit("ph cm-2 s-1") == "cm-2 s-1"
     assert standardise_unit("ct cm-2 s-1") == "cm-2 s-1"
     assert standardise_unit("cm-2 s-1") == "cm-2 s-1"
+
+def test_energy_str_formatting():
+    assert energy_str_formatting(1530e4 *u.keV) == "15.3 GeV"
+    assert energy_str_formatting(1530 *u.GeV) == "1.53 TeV"
+    assert energy_str_formatting(1530 *u.GeV) == "1.53 TeV"
+    assert energy_str_formatting(1530.5e8 *u.keV) == "153 TeV"
+
+def test_energy_unit_format():
```

Comment:
Maybe also ``pytest.parametrize``

Reply:
Thanks @nstarman. I've implemented the recommended suggestions.
Code is much more compact now.
We'll see if similar features is implemented at some point in astropy as discussed [here](https://github.com/astropy/astropy/issues/6913).